anta/cli/nrfu/utils.py

```diff
@@ -122,6 +124,14 @@ def print_jinja(results: ResultManager, template: pathlib.Path, output: pathlib.
             file.write(report)
 
 
+def save_to_csv(ctx: click.Context, csv_file: pathlib.Path | None = None) -> None:
+    """Save results to a CSV file."""
+    if csv_file is not None:
+        ReportCsv().csv_report(results=_get_result_manager(ctx), csv_filename=csv_file)
```

Comment:
More appropriate to use a classmethod to generate the report since we don't really need to keep state with an instance;

`ReportCsv.generate()` or `ReportCsv.csv_report()`, I used `MDReport.generate_report()` for the MD report. Again willing to change it but we should be consistent.

I think `CSVReport.generate()` and `MDReport.generate()` looks appropriate.

Reply:
Update code accordingly: 

```python
ReportCsv.generate(results=_get_result_manager(ctx), csv_filename=csv_file)`
```